feat: add a11y-alt-text bot

DerTimonius opened the pull request:
## PR Checklist

- [X] Addresses an existing open issue: fixes #825 
- [X] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [X] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This PR adds a new bot to check for alt text in issue, PR or discussion comments.

With the .yml file, I followed the suggestion in the [official repo](https://github.com/github/accessibility-alt-text-bot) and adapted the used version to the latest version.

JoshuaKGoldberg reviewed (changes requested):
Thanks for sending this in @DerTimonius! Quoting https://github.com/JoshuaKGoldberg/create-typescript-app/issues/825#issuecomment-1761947637:

> I'll want to see examples of the bot in another repository that you've set up with the latest version of the bot. I'd suggest using this template to make a new repository. 

In other words: could you please show this working in a repo you've created? It'll help reviewing to have us both play with it.

Also, note that you've got failures in CI. See the `.github/DEVELOPMENT.md` file for how to resolve them. Or let me know if that's unclear. Cheers!

DerTimonius commented:
Sure, I added it to a repo of mine and opened an issue with images with missing alt texts: https://github.com/DerTimonius/til-blog/issues/7

DerTimonius commented:
Tests are fixed now 🙂

JoshuaKGoldberg commented on src/steps/writing/creation/rootFiles.ts:
```diff
@@ -46,7 +46,7 @@ export async function createRootFiles(options: Options) {
 			}),
 		}),
-		".nvmrc": `20.8.1\n`,
+		".nvmrc": `20.9.0\n`,
```
#877 🫠

DerTimonius replied:
I noticed that the other error was also caused by renovate bot. I could take a look into this, but I guess this would be out of scope of this PR?

JoshuaKGoldberg commented on .github/workflows/a11y-add-alt-bot.yml:
```diff
@@ -0,0 +1,26 @@
+jobs:
+  accessibility_alt_text_bot:
+    if: ${{ github.event.issue || github.event.pull_request || github.event.discussion }}
+    name: Check alt text is set on issue or pull requests
```
[Cleanup] This repo's workflows generally omit `name` when it doesn't add much. IMO this one can be omitted.

```suggestion
```

JoshuaKGoldberg replied:
[Cleanup] This repo generally omits `name` when there's not a strong need for it. It's a bit less explicit/verbose than the suggested blurb from the bot's README.

```suggestion
```

JoshuaKGoldberg commented on .github/workflows/a11y-add-alt-bot.yml:
```diff
@@ -0,0 +1,26 @@
+    name: Check alt text is set on issue or pull requests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get action 'github/accessibility-alt-text-bot'
+        uses: github/accessibility-alt-text-bot@v1.4.0
```
```suggestion
      - uses: github/accessibility-alt-text-bot@v1.4.0
```

JoshuaKGoldberg reviewed (approved):
Looks great, thanks!

Left a few stylistic cleanups as comments - they're super minimal and not big deals at all. I can make these changes and fix up the `main` stuff. Cheers! 

<img alt="Goth girl shouting furiously in triumph" src="https://media1.giphy.com/media/h3dsteSUVxMG9YS2Ui/giphy.gif"/>

JoshuaKGoldberg reviewed (changes requested):
Oh right almost forgot - this'll need to be added to the `workflows.ts` file and its corresponding test. Otherwise this won't be added to created/templated repos.

DerTimonius commented:
@JoshuaKGoldberg I added the bot to the created files (good idea, did not think about that) - had to change a few types along the way. Also I added the .yml file to the `filesExpectedToBeChanged` Set in the `migrate-test-e2e.js` file - I wasn't sure if I should change the formatting of the original file or if I should add it to the set instead and went with the latter. I hope that's okay, if not I'll change it 🙂

JoshuaKGoldberg commented:
> Also I added the .yml file to the filesExpectedToBeChanged Set in the migrate-test-e2e.js file - I wasn't sure if I should change the formatting of the original file or if I should add it to the set instead and went with the latter. I hope that's okay, if not I'll change it 🙂

Mmhh yeah it's the other way around: the files in the repo should match the templates. I'd like to whittle that list down of _"these are intentionally different"_ to zero over time. Change request: have the `.yml` file match the template. Thanks!

DerTimonius commented:
@JoshuaKGoldberg The latest change also changes the location of the `if` statement in the .yml file. Again, I'm a bit unsure wether the `if` should be in the steps as it is now or in the same scope as `runs-on` and `steps`? I already fixed this locally, but I wanted to wait for your feedback on this one

JoshuaKGoldberg reviewed (approved):
💯. I suspect the `if` isn't necessary, but that feels like a suggestion to make on the accessibility-alt-text-bot repo. I can file ~that over there~ #997.

Otherwise, yeah, this is looking wonderful. Thanks! 🚀
